v2x/vlog_to_pbtype.py

```diff
@@ -44,7 +44,9 @@
                               with this wire
 
 The following are allowed on ports:
```

Comment:
I think this change is fine for now -- however if we want this to work with more complicated models, then we will need Yosys to understand that these ports represent clock signals and correctly propagate the result.

Can you log an issue about potentially writing a Yosys Python module which reads these properties (or looks at the wire and signal names) and make Yosys understand this is a clock signal?

In fact a lot of the attribute override functionality should probably be converted to Python Yosys passes that the outside wrapper tool only dealing with a single set of details.

Reply:
@mithro So you want to have a Yosys Python that identifies clocks? How would it inform Yosys about them? By setting eg. the `CLOCK` attribute?

Reply:
Not 100% sure. 

The solution might be multiple passes yosys passes tied together;
 - "Annotate flip flops clock inputs with clock attributes"
 - "Annotate blackbox ports named `.*clk.*` with clock attributes"
 - "Propagate clock attributes (outwards) following the logic cones"

Then the v2x tool just cares about if something has a clock attribute and doesn't care about how the thing got the clock attribute.